test/functional/test_framework/util.py

```diff
@@ -629,3 +630,37 @@ def find_vout_for_address(node, txid, addr):
         if any([addr == a for a in tx["vout"][i]["scriptPubKey"]["addresses"]]):
             return i
     raise RuntimeError("Vout not found for address: txid=%s, addr=%s" % (txid, addr))
+
+def modinv(a, n):
+    """Compute the modular inverse of a modulo n using the extended Euclidean
+    Algorithm. See https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm#Modular_integers.
+    """
+    # TODO: This does not use Python's own modular inverse pow(a, n-2, n) since
+    # that uses Fermat's little theorem which is significantly slower. However,
```

Comment:
"... uses Fermat's little theorem which is significantly slower" doesn't really make sense. Fermat's little theorem is that a^p = a (mod p). It's not an algorithm for calculating powers.

I'd suggest just removing this TODO entirely.

Reply:
`// TODO: change to pow(a, -1, n)` should be kept though.

Reply:
meh. Now that it's implemented, it's implemented. I don't see much benefit to changing it (and in general I don't think we should be leaving TODOs in the code)

Reply:
My two cents: it almost always makes sense to use what the standard library provides instead of reinventing the wheel with own implementations. Same as C++17 and C++20 substitutes will eventually be replaced when we switch over (e.g. for `std::span` or `std::optional`), the same should be done for Python functions in the future. No need to keep the codebase larger than necessary. "Now that it's implemented, it's implemented." doesn't really sound like a convincing argument to me.

Reply:
Suggest to remove the TODO and open this change as a PR or issue and discuss it there.  Then comments won't get lost, we can show benchmarks, etc.  Eventually this change will need review (with evidence) before it is merged.  We're not just going to blindly trust a TODO, so there's no value in having it.

Reply:
For now, I have kept a shortened TODO since personally I have a slight preference to keep it. While I dislike most instances of TODOs in the code, this is an ok one IMO because it is very clear what there is to do and when it should be done. I don't think it's better to open a PR now that will have to stay open for a long time until the min version is bumped. It's just a different kind of clutter in the project.

Reply:
ok!